apps/chat/src/components/Chat/RenameConversationModal.tsx

```diff
@@ -0,0 +1,231 @@
+import {
+  KeyboardEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import {
+  doesHaveDotsInTheEnd,
+  isEntityNameOnSameLevelUnique,
+  prepareEntityName,
+  trimEndDots,
+} from '@/src/utils/app/common';
+import { notAllowedSymbolsRegex } from '@/src/utils/app/file';
+
+import { ModalState } from '@/src/types/modal';
+import { Translation } from '@/src/types/translation';
+
+import {
+  ConversationsActions,
+  ConversationsSelectors,
+} from '@/src/store/conversations/conversations.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+import { UIActions } from '@/src/store/ui/ui.reducers';
+
+import { DISALLOW_INTERACTIONS } from '@/src/constants/modal';
+
+import { ConfirmDialog } from '../Common/ConfirmDialog';
+import Modal from '../Common/Modal';
+
+export const RenameConversationModal = () => {
```

Comment:
I think we don't really need `View` component here. Move this component render into `ConversationContextMenu` and use state to manage component visibility.